app/src/main/java/org/kiwix/kiwixmobile/localFileTransfer/WifiDirectManager.kt

```diff
@@ -57,13 +57,13 @@ import javax.inject.Inject
 /**
  * Manager for the Wifi-P2p API, used in the local file transfer module
  */
-@SuppressWarnings("MissingPermission")
+@SuppressWarnings("MissingPermission", "ProtectedMemberInFinalClass")
 class WifiDirectManager @Inject constructor(
   private val activity: Activity,
   private val sharedPreferenceUtil: SharedPreferenceUtil,
   private val alertDialogShower: AlertDialogShower
 ) : ChannelListener, PeerListListener, ConnectionInfoListener, P2pEventListener {
-  private val callbacks = activity as Callbacks
```

Comment:
we could also make this fragment scoped, it would be a bit complicated as you would have to create a Fragment subcomponent to be able to provide a fragment into the constructor. Want to learn a bit more about dagger? Or happy enough with how it is?

Reply:
> Want to learn a bit more about dagger?

Me:raised_hand_with_fingers_splayed: ; irrespective of this PR

Reply:
I have lots during the initial weeks of school and I want to get these merged, so I'd prefer to keep as is. That is, if you think making it fragment-scoped is complicated.

Reply:
```

  override fun onDestroyView() {
    wifiDirectManager.stopWifiDirectManager()
```
or
```

  override fun onDestroyView() {
    wifiDirectManager.callbacks = null
```
?

Reply:
or moving `callbacks = null` to onDestroyView so that we have both?

Reply:
both, use the expectation of symmetry. If you set the callbacks in onCreateView then you should remove them in onDestroyView.

The better alternative is ["Handling Lifecycles with Lifecycle-Aware Components"](https://developer.android.com/topic/libraries/architecture/lifecycle) but I can live with this

Reply:
Never heard of expectation of symmetry, that's a really nice visualization and argument.